From f45908f7194c14eccfc3abdef754bca28b46da72 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Thu, 20 Jul 2017 07:55:24 +0100
Subject: [PATCH 36/36] Add Arc-theme specific styling support

By default we expect to use Adapta theme with the installer, however it
is very likely the MATE ISO will continue to ship with the Arc Darker theme.
As such we'll detect this and override the styling to match the sidebar to
the titlebar colour under Arc Darker.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 os_installer2/data/styling.css | 6 ++++++
 os_installer2/mainwindow.py    | 6 +++++-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/os_installer2/data/styling.css b/os_installer2/data/styling.css
index 582b0eb..21a13f8 100644
--- a/os_installer2/data/styling.css
+++ b/os_installer2/data/styling.css
@@ -13,3 +13,9 @@ scrolledwindow {
     box-shadow: 0 2px 3px 1px alpha(black, 0.47);
     color: white;
 }
+
+.arc-theme .header-box {
+    background-color: #2F343F;
+    box-shadow: 0 2px 3px 1px alpha(black, 0.47);
+    color: white;
+}
diff --git a/os_installer2/mainwindow.py b/os_installer2/mainwindow.py
index 02b647a..9c10559 100644
--- a/os_installer2/mainwindow.py
+++ b/os_installer2/mainwindow.py
@@ -131,7 +131,8 @@ class MainWindow(Gtk.ApplicationWindow):
         Gtk.ApplicationWindow.__init__(self, application=app)
         self.application = app
 
-        Gtk.Settings.get_default().set_property("gtk-application-prefer-dark-theme", False)
+        settings = Gtk.Settings.get_default()
+        settings.set_property("gtk-application-prefer-dark-theme", False)
 
         self.image_step = Gtk.Image.new_from_icon_name("system-software-install", Gtk.IconSize.DIALOG)
         self.image_step.set_property("margin", 8)
@@ -221,6 +222,9 @@ class MainWindow(Gtk.ApplicationWindow):
         self.info.owner = self
 
         self.get_style_context().add_class("installer-window")
+        gtktheme = settings.get_property("gtk-theme-name").lower()
+        if gtktheme.startswith("arc"):
+            self.get_style_context().add_class("arc-theme")
 
         # Hook up actions
         self.prev_button.connect("clicked", lambda x: self.prev_page())
-- 
2.13.3

